waku/v2/node/storage/migration/migrations_scripts/message/00007_updatePrimaryKey.up.sql

```diff
@@ -0,0 +1,18 @@
+ALTER TABLE message RENAME TO message_backup;
+
+CREATE TABLE IF NOT EXISTS message(
+  pubsubTopic BLOB NOT NULL,
+  contentTopic BLOB NOT NULL,
+  payload BLOB,
+  version INTEGER NOT NULL,
+  timestamp INTEGER NOT NULL,
+  id BLOB,
+  storedAt INTEGER NOT NULL,
+  CONSTRAINT messageIndex PRIMARY KEY (storedAt, id, pubsubTopic)
+) WITHOUT ROWID;
+
+INSERT OR IGNORE INTO message(pubsubTopic, contentTopic, payload, version, timestamp, id, storedAt)
```

Comment:
Added the `OR IGNORE` clause to skip the stored messages that do not comply with the new constraint during the migration to the new scheme.

Reply:
In the local database copy that I have, with schema version 4, there were ~7000 messages that did not comply with the new constraint. With the ignore clause, those messages are dropped. For now, this is acceptable since we are not sure why those messages are not complying with the new contraint

Reply:
The Previous constraint was:
```SQL
  CONSTRAINT messageIndex PRIMARY KEY (senderTimestamp, id, pubsubTopic)
```
That constriant allowed that some messages had the same `receivedTimestamp` (now renamed to `storedAt`).

-------

With the new constraint as:
```SQL
  CONSTRAINT messageIndex PRIMARY KEY (storedAt, id, pubsubTopic)
```
While migrating and inserting the previous messages in the new table (with the new constraint), the messages with the same `storedAt` value failed; making the migration fail.

------

The question here is: _under which circumstances did those messages got inserted in the database._ 

In previous releases (v0.9 and v0.10) the `receivedTimestamp` was set to `getNanosecondTime(getTime().toUnixFloat())`. And with a nanosecond resolution, it is disturbing how the uniqueness constraint failed.